src/components/Navigation/JoinCreateLeagueBtn/JoinCreateLeague.jsx

```diff
@@ -0,0 +1,222 @@
+import React, { Component } from 'react';
+import './JoinCreateLeague.css';
+import { PlusSquare } from 'react-feather'
+import Modal from 'react-responsive-modal';
+import Select from 'react-select';
+import Button from '../../Button/Button';
+import DayPicker from 'react-day-picker';
+import 'react-day-picker/lib/style.css';
+import { withAlert } from 'react-alert'
+import { createLeague, joinLeague } from '../../../api/api';
+
+const selectOptions = [
+  { value: 'join', label: 'Join a League' },
+  { value: 'create', label: 'Create a League' }
+]
+
+class JoinCreateLeague extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isModalOpen: false,
+      selectedOption: selectOptions[0],
+      inviteCode: '',
+      name: '',
+      leagueName: '',
+      buyingPower: 0,
+      selectedStartDay: null,
+      selectedEndDay: null,
+      createLeagueErrorMsg: ''
+    };
+  }
+
+  openModal = () => {
+    this.setState({ isModalOpen: true });
+  };
+
+  closeModal = () => {
+    this.setState({ isModalOpen: false });
+  };
+
+  handleSelectChange = (selectedOption) => {
+    this.setState({ selectedOption });
+  }
+
+  handleInputChange = (event) => {
+    this.setState({
+      [event.target.id]: event.target.value
+    });
+  }
+
+  handleOnDayChange = (type, date) => {
+    let newState = {};
+    // The two variables below are used so two setState calls don't need to be called
+    let currentStart = this.state.selectedStartDay;
+    let currentEnd = this.state.selectedEndDay;
+    
+    switch(type) {
+      case 'start':
+        currentStart = date;
+        newState = {
+          selectedStartDay: date
+        }
+        break;
+      case 'end':
+        currentEnd = date;
+        newState = {
+          selectedEndDay: date
+        }
+        break;
+      default:
+        break;
+    }
+
+    // Only check date validation if both fields are filled
+    if (currentStart && 
+        currentEnd && 
+        !this.validateDates(currentStart, currentEnd)) {
+      newState['createLeagueErrorMsg'] = 'Start date cannot be before today and the end date must be after the start date.';
+      this.setState(newState);
+    }
+    else {
+      newState['createLeagueErrorMsg'] = null;
+      this.setState(newState);
+    }
+  }
+
+  joinLeagueFieldValidate = () => {
+    return this.state.inviteCode && this.state.name;
+  }
+
+  createLeagueFieldValidate = () => {
+    return this.state.name && 
+           this.state.leagueName && 
+           this.state.buyingPower && 
+           parseInt(this.state.buyingPower, 10) > 0 && 
+           this.validateDates(this.state.selectedStartDay, this.state.selectedEndDay);
+  }
+
+  validateDates = (start, end) => {    
+    return start &&
+           end &&
```

Comment:
Not really important but didn't you already check if start and end are null in lines 76 and 77?

Reply:
never mind

Reply:
i fixed it